### PR TITLE
Create Calendar page

### LINKED ISSUE #176: Populate ICAL Feed
We would like to populate the ICAL feed with events for this semester and link it on our website.

### LINKED ISSUE #189: Feed announcements into iCal feed
This will also require having part of the announcement process be removing the placeholder for the event

### LINKED ISSUE #202: Document iCal feed update process
Write docs on how to update iCal feed - as part of process for publishing talk information.
https://github.com/RITlug/ritlug.github.io/issues/189

### PR DESCRIPTION
**Pull request by ct-martin**
* Uses the iCal feed
  * Updated to include announcement posts (fixes #189)
* Created placeholder events for spring 2020 semester (fixes #176)
* Added more info to README on what the `meetings-meetups/` folder is about (fixes #202)
* Note: not currently linked to from anywhere

image

cc @jrtechs @Tjzabel @jwflory

### CODE REVIEW

File: announcements/_posts/2019-11-15-w12-kernel-tracing.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 author: Jeffery Russell (@jrtechs)
 title: "Kernel Tracing"
 layout: post-event
-date-start: "2019-10-18 16:30"
-date-end: "2019-10-18 18:00"
+date-start: "2019-11-18 16:30"
```
Comment (jrtechs) [inline announcements/_posts/2019-11-15-w12-kernel-tracing.md]: Supposed to be November 15th
kernel

Comment (ct-martin) [reply announcements/_posts/2019-11-15-w12-kernel-tracing.md]: Ah, I just looked at the month. Give me a sec and I'll fix

**Comment by ct-martin** [comment]
@jrtechs I added the calendar link to the home page (replacing Talks since those are fed into the calendar as well), does that work for you? The navbar is pretty full and it might be worthwhile to make a separate issue to talk about condensing that, maybe by adding a dropdown

**Comment by ct-martin** [comment]
When talking with @jwflory he pointed out that proper recurring events would be nice. I added an `rrule` frontmatter parameter to iCal generation which uses the iCalendar `RRULE` spec for events. Since `RRULE` allows infinite recurrence, there's an arbitrary limit of 25 occurrences.

File: README.md [hunk 2]
```diff
@@ -25,12 +25,14 @@ Assets should be stored in a folder within that category, for example, `talks/sl
 Asset file names should have enough info in the file name to know what references it (such as the associated talk date for slides) or be in a subfolder for things with a lot of assets (such as the TigerOS project).
 
 Current categories:
-* ~Announcements (`announcements/`)~ (deprecated)
+* Announcements (`announcements/`)
 * Events (`events/`)
-* Meetings & Meetups (`meetings-meetups/`) (note that these pages aren't linked from anywhere other than feeds, they are more for the metadata)
+* Meetings & Meetups (`meetings-meetups/`) (Note:  these are not pages; they are placeholder metadata for the calendar feed and should be deleted when creating the announcement post)
```
Comment (Tjzabel) [inline README.md]: what does this mean? Does this require additional metadata when creating announcement posts?

File: index.html [hunk 3]
```diff
@@ -28,7 +28,7 @@ <h2>Discuss</h2>
             <h2>Do</h2>
             <div class="list-group">
             <a href="/get-involved" class="list-group-item list-group-item-action">Get Involved</a>
-            <a href="/talks" class="list-group-item list-group-item-action">Talks</a>
+            <a href="/calendar" class="list-group-item list-group-item-action">Calendar</a>
```
Comment (Tjzabel) [inline index.html]: So does the calendar replace the Talks tab?

File: README.md (hunk 2 above)
Comment (ct-martin) [reply README.md]: The "pages" in that folder _should_ be entirely metadata and are meant to be used for the Calendar feed generation only. E.g., meetings that haven't had a topic or speaker determined yet and thus have no announcement nor talk page. These use the same metadata that the rest of the site has available, but tends to utilize some functionality more heavily, e.g. `redirect` page type. Technically you _could_ use these as real pages, but I would discourage that in favor of other sections of the site, namely, announcements, talks, and events.

File: index.html (hunk 3 above)
Comment (ct-martin) [reply index.html]: On the home page list, not the navbar. The navbar is already really full so I opted to not add more to it. I think further linking/navbar changes would be best for another issue/PR

Comment (jrtechs) [reply index.html]: the talks tab is still on the page's main navbar

**Comment by jrtechs** [comment]
@ct-martin I like these changes, as a future issue I think we could do a ritlug website 101 where we explain what needs to be done for the following things:
- planning of events (for calendar metadata)
- talk announcements
- slides for talks
- events post (things like Maker faire)

Still not fully convinced whether this should be in the run book or this repo.

**Comment by jwflory** [comment]
> Still not fully convinced whether this should be in the run book or this repo.

From my point-of-view, the answer depends if you expect this to be a task almost exclusively done by eboard (i.e. Runbook) or if you want to open it up more widely to the community (i.e. contributing guidelines for this repo).

File: index.html (hunk 3 above)
Comment (ct-martin) [reply index.html]: Yes, that was part of what I meant; this is intended behavior and not a bug

**Comment by ct-martin** [comment]
@jrtechs given that we have presenters add talks sometimes I think it should be in the repo. There has been prior conversation in issues about README vs CONTRIBUTING and so on for some of this. I think this would be better to have as an in-person conversation at FOSS Hours or RITlug. Feel free to create/update an issue and assign it to me

**Pull request merged**